guava/src/com/google/common/primitives/UnsignedBytes.java

```diff
@@ -77,7 +78,7 @@ private UnsignedBytes() {}
    *
    * @since 6.0
    */
-  public static @NonNegative int toInt(byte value) {
+  public static @NonNegative @Unsigned int toInt(@UnknownSignedness byte value) {
```

Comment:
I don't understand these annotations.

The argument is [treated as unsigned](https://github.com/typetools/guava/pull/29/files#diff-3a4acac88045451334d3d48ddbedb4a6R74), so a programmer who passed in a signed value would be surprised.  I think the formal parameter should be `@Unsigned` to reflect the intended use of the method.

The result is guaranteed to be in the constant positive range, so I think that `@SignednessGlb` would be a more appropriate return type.

Reply:
@mernst Yeah you are right. But the user can input a negative number too, the result will be ```value+256```. Should we write the formal parameter as ```@Unsigned```?

The return type should be ```@SignednessGlb```.

Reply:
The documentation is a bit confusing.  "if value is negative" is in conflict with "treated as unsigned".
I can't tell if the authors intended this routine to be called on:
 * signed numbers, with a particular behavior for negative ones
 * unsigned numbers, with simple semantics.
 * any number, which is treated as a bit value rather than as a number (the bits are re-interpreted as unsigned)

I guess we have to assume the latter, in which case you are right that the parameter should be `@Unsigned`.
My initial suggestion was based on interpretation 2.

Reply:
Sorry, typo: 

I guess we have to assume the latter, in which case you are right that the parameter should be **`@UnknownSignedness`**.

Reply:
@mernst Travis is unable to detect ```@SignednessGlb```. I think typetools/guava is running the old version of checker framework.  Need to update it.